user.go

```diff
@@ -96,6 +96,15 @@ type user struct {
 
 	networks        []*network
 	downstreamConns []*downstreamConn
+
+	pendingListsLock sync.Mutex
+	pendingLists     []pendingList
+}
+
+type pendingList struct {
+	downstreamID uint64
+	// list of pending per-upstream LIST requests and their corresponding LIST message parameters
+	pendingRequests map[int64][]string
```

Comment:
Can we replace `[]string` with the whole `*irc.Message`? It would make it clearer what the items are.

Might as well rename `pendingRequests` to `pendingCommands`?

Reply:
Right, I found it confusing as well, at first I put a typedef for that but it was kinda overkill. `*irc.Message` is a good idea.